src/visualizers-workbench/P5VisualizerTemplate.ts

```diff
@@ -81,9 +76,15 @@ class P5VisualizerTemplate extends P5Visualizer {
         // storing its value as a local constant for future reference
         const sketch = this.sketch
 
+        // get Numberscope palette colors from page style
+        const style = getComputedStyle(document.body)
+        this.nsGray400 = style.getPropertyValue('--bs-gray-400')
+        this.nsNavBG = style.getPropertyValue('--ns-navigation-background')
+        this.nsInfoBG = style.getPropertyValue('--ns-information-background')
+
```

Comment:
Whoa --

(a) Are you sure delving around in the style properties of the app webpage isn't just a little advanced/arcane for a tutorial template for a first-time Visualizer writer? And what if we switch away from bootstrap? Then `--bs-gray-400` goes away. And I get it that is nice to grab the numberscope palette from the style properties we put at the top level so that the visualizer coordinates its appearance with the overall app, but to me that belongs in a nice new `visualizer tips and tricks` doc page, not the intro.

(b) Even if we decide to delve around in the style properties in this tutorial, it seems to me that this would be a perfect opportunity to illustrate what you might _actually_ use `inhabit()` for -- there you have your hands on the containing div, and it seems to me that if you're going to extract style properties for use in the visualizer, that's the place you ought to get them from. Then the Vue app could style that div to influence the appearance of (conformant) Visualizers. But my _real_ feeling is all this ultimately will be better handled by a solution to #210.

So in sum, what's wrong in the context of this tutorial of just picking some nice CSS color names like 'DarkBlue' and 'CornflowerBlue' or something like that and using those?

Reply:
There's a subtlety in using `inhabit()` for this: if you set variables in `inhabit()` that are used in `setup()`, you have to do that before the call to `super.inhabit()`, which calls setup. Once I realized this, I decided to remove `inhabit()` from the template...

> So in sum, what's wrong in the context of this tutorial of just picking some nice CSS color names like 'DarkBlue' and 'CornflowerBlue' or something like that and using those?

As you said, a tutorial sets the tone for how people make visualizers, and beauty is a core goal. I'd like to set the example of choosing colors with care. If the CSS stuff is too confusing, maybe we could use "magic numbers" with comments explaining that they're chosen to match the page style?

Reply:
OK, symbolic color names in place. Thanks.